src/features/hotspots/settings/discovery/useDiscoveryPoll.ts

```diff
@@ -0,0 +1,115 @@
+import { Hotspot } from '@helium/http'
+import { getUnixTime } from 'date-fns'
+import { isEqual } from 'lodash'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { fetchDiscoveryById } from '../../../../store/discovery/discoverySlice'
+import { ViewState } from '../../../../store/discovery/discoveryTypes'
+import { RootState } from '../../../../store/rootReducer'
+import { useAppDispatch } from '../../../../store/store'
+
+type Props = { viewState: ViewState; hotspot: Hotspot }
+
+const unixTime = (dateStr?: string) => {
+  const insertDate = dateStr ? new Date(dateStr) : null
+  if (!insertDate) return 0
+
+  return getUnixTime(insertDate)
+}
+const useDiscoveryPoll = ({ viewState, hotspot }: Props) => {
```

Comment:
I have used https://github.com/ocetnik/react-native-background-timer in the past too for background timers. But looks like you got it working without needing it

Reply:
Nice. I'll keep that in mind for future stuff.